core/templates/components/site-search/site-search.twig

```diff
@@ -0,0 +1,26 @@
+{#
+/**
+ * @file
+ * Global Site Searach Component
+ *
+ * A default link component.
+ *
+ * Available variables:
+ * - action: Action of the form field
+ * - method: Method of the form field
+ * - input_attributes: Additional attributes for the input field
+ * - placeholder: Placeholder text for the input field
+ * - button_attributes: Additional button attributes
+ * - additional_fields: Any additional fields, e.g., hidden input fields
+ *
+ */
+#}
+<div class="su-site-search" role="search">
+ <form action="{{ action }}" method="{{ method }}" accept-charset="UTF-8">
+  <label class="su-site-search__sr-label" for="search-field">Search this site</label>
+  <input{{ input_attributes }} type="text" id="search-field" class="su-site-search__input" placeholder="{{ placeholder }}" maxlength="128">
+  <button{{ button_attributes }} type="submit" class="su-site-search__submit" aria-label="Search"><img src="../../../img/icon-search.svg" alt="Search button"/></button>
```

Comment:
How are we handling paths to images? Is it okay to use relative paths within Decanter?

Also, I can imagine some sites might want to use a different icon, or have no icon. Should the icon be a parameter, and / or optional?

Reply:
Hmm....this might get a little tricky since I'm using the icon as the submit button here and a submit button is recommended for accessibility. Or maybe we can do a variant with a text button or no button (is this ok - might need to ask Jiatyan)?

Reply:
The path and image is a great question, and I think I'll make the img src to be a variable so people have the option to use a different icon at least. Thanks.